pandas/core/frame.py

```diff
@@ -2014,36 +2027,65 @@ def to_dict(
             )
 
         elif orient == "split":
-            return into_c(
-                (
-                    ("index", self.index.tolist()),
-                    ("columns", self.columns.tolist()),
+            if index:
+                return into_c(
                     (
-                        "data",
-                        [
-                            list(map(maybe_box_native, t))
-                            for t in self.itertuples(index=False, name=None)
-                        ],
-                    ),
+                        ("index", self.index.tolist()),
+                        ("columns", self.columns.tolist()),
+                        (
+                            "data",
+                            [
+                                list(map(maybe_box_native, t))
+                                for t in self.itertuples(index=False, name=None)
+                            ],
+                        ),
+                    )
+                )
+            else:
+                return into_c(
+                    (
+                        ("columns", self.columns.tolist()),
+                        (
+                            "data",
+                            [
+                                list(map(maybe_box_native, t))
+                                for t in self.itertuples(index=False, name=None)
+                            ],
+                        ),
```

Comment:
Sorry the diff is a bit difficult to read with the change of indentation, but if I'm not wrong, this is duplicating all this code, and I think it can be easily prevented using this pattern:
```
(('index',) if index else ()) + ('columns', 'data')
```

Does this make sense to you?

Reply:
Yes, the diff on github looks more messy than on vscode. I've applied the change.